src/BenchmarkDotNet.Core/Analysers/MultimodalDistributionAnalyzer.cs

```diff
@@ -0,0 +1,37 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using BenchmarkDotNet.Engines;
+using BenchmarkDotNet.Extensions;
+using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Mathematics;
+using BenchmarkDotNet.Reports;
+using JetBrains.Annotations;
+
+namespace BenchmarkDotNet.Analysers
+{
+    public class MultimodalDistributionAnalyzer : AnalyserBase
+    {
+        public override string Id => "MultimodalDistribution";
+        public static readonly IAnalyser Default = new MultimodalDistributionAnalyzer();
+
+        private MultimodalDistributionAnalyzer() { }
+
+        [NotNull]
+        private Conclusion Create([NotNull] string kind, double mValue, [CanBeNull] BenchmarkReport report)
+            => CreateWarning($"It seems that the distribution {kind} (mValue = {mValue})", report);
+
+        public override IEnumerable<Conclusion> AnalyseReport(BenchmarkReport report, Summary summary)
+        {
+            var statistics = report.ResultStatistics;
+            if (statistics == null || statistics.N < 15)
```

Comment:
Yes. Unfortunately, it's too hard to detect multimodal distribution with the small amount of iteration and keep the false positive rate low.
I choose 15 here because we run at least 15 iterations by default. If user don't customize amount of iteration, the analysis will be applied.

Reply:
thanks for the explanation! we should move 15 to some const then to make it more obvious (what is magic number, what is part of a smart heuristic ;p )

Reply:
Yes, will do it. Probably we have to introduce `MagicNumbers.cs`. =)